src/sorting/gnome_sort.rs

```diff
@@ -31,30 +31,30 @@ mod tests {
     #[test]
     fn basic() {
         let res = gnome_sort(&[6, 5, -8, 3, 2, 3]);
-        assert_eq!(res, vec![-8, 2, 3, 3, 5, 6]);
+        assert!(crate::sorting::is_sorted(&res));
```

Comment:
The checks are not equivalent. `[1, 2, 3]` would pass the second check, but it's clearly not what we want.
Please also add a check that the resulting array contains all the same elements as the input array.

Reply:
```rust
    fn basic() {
        let res = gnome_sort(&[6, 5, -8, 3, 2, 3]);
        // assume that res is [1, 2, 3]
        assert!(crate::sorting::is_sorted(&[1, 2, 3]));
        // the test passes
}
```

Reply:
Currently, `is_sorted` doesn't check that the resulting array contains all the same elements as the input array

Reply:
I have committed new code with the checks for members - kindly review, thanks